content/tutorials/spatial_transform_crs_2/index.Rmd

```diff
@@ -0,0 +1,495 @@
+See e.g.
+[Wikipedia](https://nl.wikipedia.org/wiki/Rijksdriehoeksco%C3%B6rdinaten) for more background about this CRS.
+
+Essentially we are dealing with a different projected CRS than `EPSG:31370`.
+`EPSG:28992` uses the 'Amersfoort' geodetic datum, defined using the 'Bessel 1841' ellipsoid, while `EPSG:31370` has the 'Reseau National Belge 1972' datum, defined using the 'International 1924' ellipsoid [^1].
+
+[^1]: Note that several European countries, including Belgium, now provide national projected CRSs that use the common ETRS89 datum (European Terrestrial Reference System 1989).
+    An example is the Belgian CRS `EPSG:3812` (`r st_crs(3812)$Name`).
+
+We will transform `points` (back) to `EPSG:31370` since the data are outside the intended area of usage.
+Note that this dataset has been made for mere demonstrative purposes:
+since at least a part of the points fall outside the area of usage (geographic bounding box) of the '`r st_crs(points)$Name`' CRS, the dataset is not appropriate to start from.
+But let's suppose it is all one had available to work with!
+
+### Standard approach
+
+By default, the appropriate transformation pipeline (concatenated coordinate operations) can differ between different areas that are represented in a spatial dataset.
+This is taken care of automatically based on the bounding boxes of each CRS, but it will also depend on the availability of specific transformation grids.
+
+```{r}
+points_31370 <- st_transform(points, "EPSG:31370")
+```
+
+Notice the completely different coordinate ranges between both CRSs:
+
+```{r collapse=TRUE}
+st_geometry(points)
+```
+
+
+```{r collapse=TRUE}
+st_geometry(points_31370)
+```
+
+A specific area shown with `ggplot2` in `EPSG:31370`:
+
+```{r ggplot_water_points}
+ggplot() + 
+  geom_sf(data = watercourses, fill = "lightblue", colour = "lightblue") +
+  geom_sf(data = points_31370, fill = "red", shape = 21) +
+  lims(x = c(154e3, 155e3), y = c(218e3, 219e3)) +
+  coord_sf(datum = "EPSG:31370")
+```
+
+### Enforcing a specific pipeline
+
+We can request available transformation pipelines with `sf_proj_pipelines()`:
+
+```{r}
+pipelines <- sf_proj_pipelines("EPSG:28992", "EPSG:31370")
+```
+
+One can further limit this by specifying the area of interest (`aoi` argument), which we didn't do here.
+
+It is a dataframe with following structure:
+
+```{r}
+glimpse(pipelines)
+```
+
+Some metadata about the pipelines:
+
+```{r paged.print=FALSE}
+pipelines %>% 
+  as_tibble %>% 
+  select(definition, description, grid_count, accuracy, instantiable)
+```
+
+We can see that the first one has best accuracy.
+Several pipelines are based on one or two grids for a better accuracy.
+
+When printing the `pipelines` object, its first row is printed in a more readable manner:
+
+```{r}
+pipelines
+```
+
+Since the rows of pipelines are sorted by `instantiable` and `accuracy`, it makes sense to designate the printed pipeline as _Best instantiable operation_.
```

Comment:
How does `st_transform()` handle the choice of pipeline? Will it choose the first row of `pipelines` by default?

Reply:
I think it's not documented all too well on the R side; my statements are partially from observation. Plus, PROJ versions continue to evolve.

In R, best documentation seems at https://keen-swartz-3146c4.netlify.app/sf.html#pipelines, but it doesn't handle area-wise choices (was thinking of discussing or adding that over there, some time). It's not really discussed within `sf` docs I think.

The documentation for choosing between pipelines is at https://proj.org/operations/operations_computation.html#filtering-and-sorting-of-coordinate-operations, this is for PROJ 8.0.0 while `sf` is typically still compiled with PROJ 7.2.1 (since it's the version  at ubuntugis-unstable PPA with binaries for Linux). Most likely things have evolved already somewhat compared to 7.2.1.

From observation, it seems that this prioritisation is carried out point-wise (or area-wise), since different choices were applied by default through `st_transform()`, depending on the area at hand (outside vs. inside 'BBOX' of the CRS's 'USAGE').

Reply:
Interesting. It would be good to clarify (and maybe illustrate) that `st_transform` picks the most accurate transformation (inside bbox) or the most appropriate (ballpark; outside bbox) depending on the situation.

Reply:
A little more explanation has been added with 3d815734.